x/staking/types/codec.go

```diff
@@ -14,11 +14,10 @@ func RegisterCodec(cdc *codec.Codec) {
 }
 
 // generic sealed codec to be used throughout this module
-var ModuleCdc *codec.Codec
+var ModuleCdc = codec.New()
```

Comment:
Does this mean that it is possible that someone can use the codec although it has not been registered by calling `init`?
Looking at the code, it seems to me that `init` was ensuring that ModuleCdc is nil unless the code has been instantiated, registered and sealed.

Reply:
> Does this mean that it is possible that someone can use the codec although it has not been registered by calling `init`?

This is not generally the case because when you import a module the `init` function is always called before any other code in the library or the library-user package. however it turns out that this may be a possibility within _another_ init function from a different module. The order of init calls is not totally, clear so this may be an attack vector on the ModuleCdc... just to be safe, I'll define everything in init and then set the ModuleCdc to that. (as was happening in some places previously)

Thanks for bringing this up

Reply:
`init` is an antipattern fwiw _ducks_

Reply:
🥚 🍅  🍆